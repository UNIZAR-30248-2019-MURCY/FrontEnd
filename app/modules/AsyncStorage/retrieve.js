import { AsyncStorage } from 'react-native';

export const retrieveItem = async (key) => {
    let data ='';
    try {
        data = await AsyncStorage.getItem(key)
        console.log(data)
    } catch (error) {
        console.log(error.message);
    }
    return data;
}
