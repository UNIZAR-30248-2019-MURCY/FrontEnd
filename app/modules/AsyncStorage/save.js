import { AsyncStorage } from 'react-native';

export const saveData = async (key, data) => {
    try {
        await AsyncStorage.setItem(key, JSON.stringify(data));
        console.log('data stored');
    } catch (error) {
        // Error retrieving data
        console.log(error.message);
    }
};
