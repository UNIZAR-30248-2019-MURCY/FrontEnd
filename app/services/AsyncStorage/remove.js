import { AsyncStorage } from 'react-native';

export const removeItem = async (key) => {
    try {
        await AsyncStorage.removeItem(key);
        console.log('data removed');
        return true;
    } catch (error) {
        console.log(error.message);
        return false;
    }
}
