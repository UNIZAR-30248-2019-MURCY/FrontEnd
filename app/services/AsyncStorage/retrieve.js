import { AsyncStorage } from 'react-native';

export const retrieveItem = async (key) => {
    let data ='';
    try {
        let dataStored = await AsyncStorage.getItem(key)
        data = JSON.parse(dataStored).token;
        console.log('retrievedItem')
    } catch (error) {
        console.log(error.message);
    }
    return data;
}
