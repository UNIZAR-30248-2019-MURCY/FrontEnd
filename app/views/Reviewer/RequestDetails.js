import React, {Component} from 'react';
import {AsyncStorage, FlatList, StyleSheet, TextInput, View,} from 'react-native'
import {Button, ListItem, Text} from 'react-native-elements';

export default class RequestDetails extends Component {

    constructor(props) {
        super(props);
        this.state = {
            workflow: false
        }
    }

    componentDidMount() {
        if (this.props.navigation) {
            this.setState({workflow: this.props.navigation.getParam('workflow', 'default value')});
        }
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.navigation.state.params.lastName) {
            this.setState({lastName: nextProps.navigation.state.params.lastName});
        }
    }

    render() {
        let showDetails = (
            this.state.workflow ?
                <View style={styles.containerRequest}>
                    <Text h4 style={styles.containerSubTitle}>{this.state.workflow.title}</Text>
                    <Text style={styles.containerRequestDetails}>
                        Description: {this.state.workflow.description}
                        {'\n'}{'\n'}
                        Status: {this.state.workflow.status}
                    </Text>
                    <View style={styles.containerButtons}>
                        <Button
                            buttonStyle={styles.button}
                            title="Accept"
                            onPress={() => {
                                this.props.navigation.replace('RequestActionConfirm', { type: 'accepted' });
                            }}/>

                        <Button
                            buttonStyle={styles.buttonDen}
                            title="Deny"
                            onPress={() => {
                                this.props.navigation.replace('RequestActionConfirm', { type: 'denied' });
                            }}/>
                    </View>
                </View> :
                <View></View>

        );

        return (
            <View style={styles.container}>
                <View style={styles.containerTitle}>
                    <Text h2>Details</Text>
                </View>
                <View style={styles.containerRequest2}>
                    {showDetails}
                    <View style={styles.containerRequest3}>
                        <Button
                            className='return-button'
                            type="clear"
                            buttonStyle={styles.button2}
                            title="Return"
                            titleStyle={{color: 'grey'}}
                            onPress={() => {
                                this.props.navigation.goBack();
                            }
                            }/>
                    </View>
                </View>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 20,
        //justifyContent: 'center',
    },
    containerTitle: {
        alignItems: 'center',
        marginTop: 50,
        padding: 20
    },
    containerSubTitle: {
        marginTop: 25
    },
    containerRequest: {
        flex: 1,
        alignItems: 'center',
    },
    containerRequest2: {
        flex: 1,
    },
    containerRequest3: {
        alignItems: 'center',
    },
    containerRequestDetails: {
        flex: 1,
        marginTop: 25,
        fontSize: 18,
        color: 'grey',
        textAlign: 'center',
    },
    button: {
        width: 150,
        height: 55,
        marginTop: 20,
        margin: 10,
        backgroundColor: 'grey',
        borderRadius: 14
    },
    buttonDen: {
        width: 150,
        height: 55,
        marginTop: 20,
        margin: 10,
        backgroundColor: 'red',
        borderRadius: 14
    },
    button2: {
        marginBottom: 10,
        width: 150,
        height: 55,
        borderRadius: 14
    },
    containerButtons: {
        flexDirection: 'row'
    },

})
