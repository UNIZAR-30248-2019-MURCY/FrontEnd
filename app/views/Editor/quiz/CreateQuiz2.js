import React, {Component} from 'react';
import {
    View,
    StyleSheet,
    TextInput,
    ScrollView, ActivityIndicator, FlatList, TouchableOpacity
} from 'react-native'
import {Button, Text, CheckBox} from 'react-native-elements';
import {retrieveItem} from "../../../modules/AsyncStorage/retrieve";
import {createQuiz} from "../../../services/quiz/quizFuncs";
import SwitchSelector from 'react-native-switch-selector';
import {listQuestions} from "../../../services/question/questionFuncs";
import Icon from 'react-native-vector-icons/FontAwesome';




export default class CreateQuiz extends Component {

    constructor(props) {
        super(props);
        this.state = {
            token: '',
            title: '',
            description: '',
            loading: false,
            publish: false,
            dataSource: []
        }
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    componentDidMount() {
        if (this.props.navigation) {
            if(this.props.navigation.getParam('data')){
                this.setState({title: this.props.navigation.getParam('title')})
                this.setState({description: this.props.navigation.getParam('description')})
                this.setState({dataSource: this.props.navigation.getParam('data')})
                this.setState({token: this.props.navigation.getParam('token')})
            }
        }
    }

    handleSubmit() {

        this.setState({loading: true})
        this.setState({error: false})
        console.log(this.state.title)
        console.log(this.state.description)
        console.log(this.state.dataSource)
        var questionIds= []
        this.state.dataSource.map(function(element, indice){
            questionIds.push(element.id)
        })
        console.log(questionIds)

        createQuiz(this.state.title, this.state.description, questionIds, this.state.publish ,this.state.token)
            .then((data) => {
                console.log(data);
                this.setState({loading: false})
                this.props.navigation.replace('QuizConfirm', {type: 'created'});
            })
            .catch((error) => {
                this.setState({error: error.message})
                this.setState({loading: false})
            });
    }

    FlatListItemSeparator = () => <View style={styles.line} />;


    renderItem = data =>
        <Text style={[styles.list, styles.lightText]}>  {data.item.title} </Text>


    render() {

        let showErr = (
            this.state.error ?
                <View style={styles.error} className='errorShow'>
                    <Text style={{color: 'red'}}>
                        {this.state.error}
                    </Text>
                </View> :
                <View></View>
        );
        let showLoading = (
            this.state.loading ?
                <View style={[styles.containerLoading]} className='loadingShow'>
                    <ActivityIndicator animating={this.state.loading} size="large" color="grey"/>
                </View> :
                <View></View>
        );

        return (
            <ScrollView>
                <View style={styles.container}>
                    <View style={styles.containerTitle}>
                        <Text h2>New Quiz</Text>
                    </View>
                    <View style={styles.containerCreate}>
                        <Text style={styles.subTitle}>Title and description</Text>
                        <View style={styles.container3}>

                            <Text style={[styles.list, styles.lightText]}>{this.state.title}</Text>
                            <Text style={[styles.list, styles.lightText]}>{this.state.description}</Text>
                        </View>

                        <Text style={styles.subTitle2}>Questions</Text>

                        <ScrollView>
                        <View style={styles.container2}>
                            <FlatList
                                data={this.state.dataSource}
                                ItemSeparatorComponent={this.FlatListItemSeparator}
                                renderItem={item => this.renderItem(item)}
                                keyExtractor={item => item.id.toString()}
                                extraData={this.state}
                            />
                        </View>
                        </ScrollView>


                        <View style={styles.containerSelector}>
                            <SwitchSelector
                                initial={0}
                                onPress={value => this.setState({ publish: value })}
                                textColor='grey'
                                selectedColor='white'
                                buttonColor='grey'
                                borderColor='grey'
                                hasPadding
                                options={[
                                    { label: "Draft", value: "false", },
                                    { label: "Public", value: "true",  }
                                ]}
                            />
                        </View>

                        {showErr}
                        {showLoading}

                        <Button
                            buttonStyle={styles.button}
                            className='create-quiz-button'
                            title="Create"
                            onPress={() => {
                                this.handleSubmit()
                            }}/>
                        <Button
                            className='return-button'
                            type="clear"
                            buttonStyle={styles.button2}
                            title="Return"
                            titleStyle={{color: 'grey'}}
                            onPress={() => {
                                this.props.navigation.goBack();
                            }
                            }/>
                    </View>

                </View>
            </ScrollView>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        padding: 20
    },
    containerTitle: {
        marginTop: 50,
        padding: 20
    },
    containerCreate: {
        flex: 1,
        alignItems: 'center',
        padding: 20
    },
    subTitle: {
        flex: 1,
        fontSize: 22,
        color: 'grey',
    },
    subTitle2: {
        flex: 1,
        fontSize: 22,
        color: 'grey',
        marginTop: 30,
    },
    subTitle3: {
        flex: 1,
        fontSize: 17,
        color: 'grey',
        marginBottom: 5
    },
    subTitle4: {
        flex: 1,
        fontSize: 17,
        color: 'grey',
        justifyContent: "flex-start",
        alignItems: "flex-start",
    },

    checkBoxC: {
        padding: 0,
        marginTop: 25,
        alignItems: 'center',
        backgroundColor: "white",
        borderColor: "white"
    },
    logo: {
        margin: 30,
        marginBottom: 15,
        padding: 8,
        width: 100,
        height: 100
    },
    input: {
        width: 300,
        height: 55,
        backgroundColor: 'whitesmoke',
        margin: 10,
        padding: 8,
        color: 'grey',
        borderRadius: 14,
        fontSize: 18,
        fontWeight: '500',
    },
    inputAns: {
        width: 260,
        height: 55,
        backgroundColor: 'whitesmoke',
        margin: 10,
        padding: 8,
        color: 'grey',
        borderRadius: 14,
        fontSize: 18,
        fontWeight: '500',
    },
    button: {
        width: 200,
        height: 50,
        marginTop: 30,
        margin: 10,
        backgroundColor: 'grey',
        borderRadius: 14
    },
    button2: {
        width: 150,
        height: 55,
        borderRadius: 14
    },
    containerButton: {
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
    },
    buttonContainer: {
        alignItems: 'center',
        justifyContent: 'center',
    },
    buttonAdd: {
        width: 140,
        height: 45,
        marginTop: 25,
        margin: 10,
        backgroundColor: '#4b7bec',
        borderRadius: 14
    },
    buttonRemove: {
        width: 140,
        height: 45,
        marginTop: 25,
        margin: 10,
        backgroundColor: '#ff5252',
        borderRadius: 14
    },
    containerLoading: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        padding: 10
    },
    containerButtons: {
        flexDirection: 'row'
    },
    error: {
        marginTop: 40,
        alignItems: 'center',
    },
    containerSelector: {
        flex: 1,
        marginTop: 40,
        marginBottom: 15,
        width: 250,
    },

    container2: {
        flex: 1,
        //backgroundColor: "#192338",
        marginTop: 20,
        marginBottom: 5,
        position: "relative",
        width: 300,
        maxHeight: 200,
    },
    container3: {
        flex: 1,
        //backgroundColor: "#192338",
        marginTop: 10,
        position: "relative",
        width: 300,
    },
    list: {
        paddingVertical: 5,
        margin: 3,
        flexDirection: "row",
        //backgroundColor: "#192338",
        justifyContent: "flex-start",
        alignItems: "flex-start",
        zIndex: -1,
        borderRadius: 14
    },
    lightText: {
        color: 'grey',
        width: 200,
        paddingLeft: 15,
        fontSize: 17
    },
    line: {
        height: 0.1,
        width: "50%",
        backgroundColor:"white"
    },
})
