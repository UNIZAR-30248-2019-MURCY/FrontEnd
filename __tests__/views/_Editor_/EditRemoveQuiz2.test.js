import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import EditRemoveQuiz2 from "../../../app/views/Editor/quiz/EditRemoveQuiz2";
import {retrieveItem} from "../../../app/modules/AsyncStorage/retrieve";
import RequestDetails from "../../../app/views/Reviewer/RequestDetails";
import EditRemoveQuiz from "../../../app/views/Editor/quiz/EditRemoveQuiz";
import CreateQuiz2 from "../../../app/views/Editor/quiz/CreateQuiz2";

describe('<EditRemoveQuiz2 />', () => {
    it('CreateQuiz renders without crashing', () => {
        const rendered = renderer.create(<EditRemoveQuiz2/>).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('CreateQuiz test against snapshot', () => {
        const welcomePage = renderer.create(<EditRemoveQuiz2/>).toJSON();
        expect(welcomePage).toMatchSnapshot();
    });
    it('handleSubmit: Edit Quiz', () => {
        let wrapper = shallow(<EditRemoveQuiz2/>);
        wrapper.instance().handleSubmit = jest.fn();
        wrapper.update();
        wrapper.find('.save-button').simulate('press');
        expect(wrapper.instance().handleSubmit).toBeCalledTimes(1);
    });
    it('goBack and navigate to QuizzesEdit', () => {
        const navigationMock = {
            goBack: jest.fn(),
            getParam: jest.fn(),
        };
        const wrapper = shallow(<EditRemoveQuiz2 navigation={navigationMock}/>);
        wrapper.find('.return-button').simulate('press');
        expect(navigationMock.goBack).toHaveBeenCalled()
    });
    it('should render the Error component if state.error is true', () => {
        let createComponent = shallow(<EditRemoveQuiz2/>);
        createComponent.setState({error: true});
        expect(createComponent.find('.errorShow').length).toBe(1);
    });
    it('should render the Loading component if state.loading is true', () => {
        let createComponent = shallow(<EditRemoveQuiz2/>);
        createComponent.setState({loading: true});
        expect(createComponent.find('.loadingShow').length).toBe(1);
    });
    it('showPublish', () => {
        let wrapper = shallow(<EditRemoveQuiz2/>);
        wrapper.instance().setState = jest.fn();
        wrapper.update();
        wrapper.find('SwitchSelector').simulate('press', true);
        expect(wrapper.instance().setState).toBeCalledWith({publish: true});

    });
    describe('.renderItemIsSelected', () => {
        const mockItem = {
            title: 'Title',
        }
        let renderItemShallowWrapper;
        let wrapper = shallow(<EditRemoveQuiz2/>);

        beforeAll(() => {
            let RenderItem = wrapper.find('.flatList').prop('renderItem');
            renderItemShallowWrapper = shallow(<RenderItem item={mockItem} />);
        });
        it('should match the snapshot', () => {
            expect(renderItemShallowWrapper).toMatchSnapshot();
        });
    });
    it('get Params Draft', async () => {
        await retrieveItem();
        const mockData = {
            title: 'Title',
            description: 'd',
            data: [],
            token: 't',
            lastWorkflow: {
                status: 'DRAFT'
            }
        }
        const navigationMock = {
            navigate: jest.fn(),
            getParam: jest.fn((param) => mockData),
        };

        const wrapper = shallow(<EditRemoveQuiz2 navigation={navigationMock} />);
        wrapper.update();
        expect(navigationMock.getParam).toHaveBeenCalled()
    });
    it('get Params Public', async () => {
        await retrieveItem();
        const mockData = {
            title: 'Title',
            description: 'd',
            data: [],
            token: 't',
            lastWorkflow: {
                status: 'PUBLIC'
            }
        }
        const navigationMock = {
            navigate: jest.fn(),
            getParam: jest.fn((param) => mockData),
        };

        const wrapper = shallow(<EditRemoveQuiz2 navigation={navigationMock} />);
        wrapper.update();
        expect(navigationMock.getParam).toHaveBeenCalled()
    });


});


