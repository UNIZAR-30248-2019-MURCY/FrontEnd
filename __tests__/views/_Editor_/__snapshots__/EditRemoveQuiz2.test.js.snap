// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditRemoveQuiz2 /> .renderItemIsSelected should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`<EditRemoveQuiz2 /> CreateQuiz test against snapshot 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": 50,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 42.5,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          Quiz
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "flex": 1,
              "fontSize": 22,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          Title and description
        </Text>
        <View
          style={
            Object {
              "flex": 1,
              "marginTop": 10,
              "position": "relative",
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "alignItems": "flex-start",
                "borderRadius": 14,
                "color": "grey",
                "flexDirection": "row",
                "fontSize": 17,
                "justifyContent": "flex-start",
                "margin": 3,
                "paddingLeft": 15,
                "paddingVertical": 5,
                "width": 200,
                "zIndex": -1,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "alignItems": "flex-start",
                "borderRadius": 14,
                "color": "grey",
                "flexDirection": "row",
                "fontSize": 17,
                "justifyContent": "flex-start",
                "margin": 3,
                "paddingLeft": 15,
                "paddingVertical": 5,
                "width": 200,
                "zIndex": -1,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "grey",
              "flex": 1,
              "fontSize": 22,
              "marginTop": 30,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          Questions
        </Text>
        <RCTScrollView>
          <View>
            <View
              style={
                Object {
                  "flex": 1,
                  "marginBottom": 5,
                  "marginTop": 20,
                  "maxHeight": 200,
                  "position": "relative",
                  "width": 300,
                }
              }
            >
              <RCTScrollView
                ItemSeparatorComponent={[Function]}
                className="flatList"
                data={Array []}
                disableVirtualization={false}
                extraData={
                  Object {
                    "data": Array [],
                    "dataSource": Array [],
                    "description": "",
                    "loading": false,
                    "publish": false,
                    "showPublish": true,
                    "title": "",
                    "token": "",
                  }
                }
                getItem={[Function]}
                getItemCount={[Function]}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={1}
                onContentSizeChange={[Function]}
                onEndReachedThreshold={2}
                onLayout={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={Array []}
                updateCellsBatchingPeriod={50}
                viewabilityConfigCallbackPairs={Array []}
                windowSize={21}
              >
                <View />
              </RCTScrollView>
            </View>
          </View>
        </RCTScrollView>
        <View />
        <View />
        <View
          style={
            Object {
              "flex": 1,
              "marginBottom": 15,
              "marginTop": 40,
              "width": 250,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {},
              ]
            }
          >
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                onLayout={[Function]}
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 50,
                    "height": 40,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "borderColor": "grey",
                      "borderRadius": 50,
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "color": "white",
                            "fontSize": 14,
                            "fontWeight": "normal",
                            "textAlign": "center",
                          },
                          Object {},
                        ]
                      }
                    >
                      Draft
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "color": "grey",
                            "fontSize": 14,
                            "fontWeight": "normal",
                            "textAlign": "center",
                          },
                          Object {},
                        ]
                      }
                    >
                      Public
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderRadius": 14,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "grey",
                  "borderColor": "#2089dc",
                  "borderRadius": 14,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "center",
                  "margin": 10,
                  "marginTop": 30,
                  "padding": 8,
                  "width": 200,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Save
              </Text>
            </View>
          </View>
        </View>
        <View />
        <View
          style={
            Object {
              "borderRadius": 14,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 14,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 55,
                  "justifyContent": "center",
                  "padding": 8,
                  "width": 150,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "grey",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Return
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
